### DISCUSSION TITLE
Opensource bait and switch?

### DISCUSSION THREAD

**Discussion by MayurVirkar** [Budibase/budibase#13411] [Help] [answered]
Hello,
I don't mind creators keeping a few features for paid versions. Of course, everyone needs to earn. I totally get that.
But restricting open-source, community-edition, crowd-contributed versions to just 20 users? That, too the is self-hosted?
That doesn't seem fair.

**Comment by aptkingston** [comment, marked as answer]
Hi @MayurVirkar.

If you've had so much success with Budibase that you have more than 20 users who use it actively, I think it's perfectly reasonable to think that Budibase is providing you with enough value that it might be time to consider giving something back.

If you're looking to trial an app with a much larger user base then you're welcome to create a public app, which won't need user accounts for end users.

We do also offer discounts for eligible nonprofit organizations if that's relevant to you.

> That doesn't seem fair.

I would argue that getting a fully functional low-code app builder for 20 users, with unlimited apps, unlimited datasources, SSO and RBAC for abolutely nothing is more than fair - it's generous 😄

**Reply by JCV67** [reply]
The issue isn't related to cost; it's that the rules of the game changed midway. I conducted extensive research before settling on Budibase, in search of an open-source, self-hosted, low-code platform. I read a lot and even recall listening to a podcast where a Budibase founder mentioned their ambition to make Budibase the WordPress of low-code platforms (WP is free after all these years). I was impressed, invested my time in it, and then one day, the rules changed. While I adore Budibase, this experience has left me with a somewhat bitter feeling. I believe that removing the user number limitation would be a fair gesture towards older users. Also, we are always convincing others about the marvelous if Budibase.

**Reply by MayurVirkar** [reply]
I am in total agreement with you. We also did extensive research.
But it's not only the user limit that I am worried about now. Also many features which made Budibase, "Budibase" have been moved to commercial plans.
This experience has made it hard for me to trust open-source projects these days.

I am not against charging for hosting, my all means go ahead. This is what almost every open-source project does, but limiting features and users on self-hosted open-source project, which my friends have contributed to, is unethical.

**Reply by aptkingston** [reply]
@MayurVirkar I don't believe we've ever moved features from free plans to paid - we've designed certain features deliberately for paid plans (backups, snippets, audit logs, custom branding, sync automations etc). The only change we made to free plans was the user count limit and the additional of branding, to encourage upgrading to paid plans when you are successful with the platform and need to scale. Which features are you referring to?

Some stats as well in case you're interested - about 98% of the commits to this repository are Budibase authored. Over 1% of the remaining commits are by a single contributor, leaving <1% of commits to all other community contributions. Actual lines of code contributions are an order of magnitude less than that again. We welcome community contributions, but are definitely not a community-build product.

With that in mind, I don't believe it's fair to call it unethical to impose a (reasonably high) limit on users on the free plan. Expecting to get every single piece of software fully functional, for free, just because you host it yourself is not realistic. I "self-host" Adobe Photoshop, but I certainly don't expect it to be free just because it runs on my computer rather than Adobe's servers.

Being open source provides transparency and trust for people who want to run, contribute to or modify our software, but it definitely does not mean we must give away everything for free.

**Reply by hamzafarooq951** [reply]
@aptkingston I agree with your points about the deliberate design of features for paid plans and the need to encourage upgrading for scalability. A software company needs to sustain itself and grow. However, I believe the issue here is more about how the software is presented rather than its pricing model.

Open-source software is typically associated with being freely available for use, modification, and distribution. While it's perfectly acceptable to have paid plans and restrict certain features, claiming to be open source when the majority of the development is done by the company and community contributions are minimal can be misleading.

In the case of Adobe Photoshop, it's clear that it's a proprietary software with a paid license. Users understand this distinction. For Budibase, it might be more transparent to describe the software as "source available" to avoid confusion and maintain trust with users.

**Reply by XeroxDev** [reply]
Sorry for "intervening" but I've to agree with the users here. I just found this project and invested time searching about it. The OSS self hosted phrased catched me and I went through all of your docs. I even considered recommending the cloud plans to my CEO, so we can internally build nice tools for over 200 users. But I wanted to test it out first and installed it on my server just to find out about this bait & switch.

If we look at the usage page on a self hosted instance
![image](https://github.com/Budibase/budibase/assets/10864620/de483613-9053-476c-bccc-796c39bd1404)

there's the 20 user limit. I don't really care about it, because I gonna use it as a single person. But what I care about: Even in your docs the images aren't updated.
![image](https://github.com/Budibase/budibase/assets/10864620/2d09fdec-a372-4a90-a9b9-efe1d78ed312)

And all the pay-walled features as well. You promised and ADVERTISED an self hosted OSS free product. You even mentioned and I quote: 
![image](https://github.com/Budibase/budibase/assets/10864620/4ddc534a-d0fc-45ea-9f17-123b39618322)
still today in the main/default branch. Or does this mean we can and are allowed to remove all the restrictions our selves? (Because it seems the restrictions are in some arbitrary / closed source / obfuscated files, so even if we would be allowed, you intentionally made it hard and unfriendly to remove those)

I really don't care about the restrictions, they're fine. It would even be okay to have to pay to remove branding (because credits should be given, no matter what). It's okay that you want to earn money but:
1. Requesting the same amount for the self hosted and cloud variant, despite you don't have to pay for others self hosted instances
2. The misleading advertisement (which is even ILLEGAL in a lot of countries (the one I'm living in as example. You would get fined up to 6 figures here or even jail time for such things))
3. This is extremely shady behavior I don't like.

and those are the reason why I won't recommend it and even gonna put it on a blacklist for future ideas.

Instead of restricting the self hosted variant and putting existing things behind a paywall, you should've just created another business version for it as a fork or just request money if we're using your cloud services. 

But because you seem to already have decided to go this path, at least have enough courage to put it everywhere that it's not a free open source (forever) project anymore but "source available" premium software, please send money.

I can't even understand why users (and groups etc) are restricted on self hosted, because you have **literally no downsides** of it being unlimited.

**Reply by JCV67** [reply]
This discussion shouldn't have been closed. We are right, you did not keep your promise. As I said before, I also did a lot of research before choosing Budibase and after that I spent many hours on the learning curve. I like budibase and I'm not in position to drop it. You should think it further and change your decision. You can always change the users limit number and drop functions restrictions. Let's say a 100 users limit on self hosted, it would still only benefit small companies and it would be good for your discussion forums. We are the ones who are fast applying new updates and the ones that like to spread the word.

**Reply by aptkingston** [reply]
@XeroxDev you're right that the screenshot you provided is outdated - we'll fix that. But that absolutely is not representative of our "advertising" of plan limits. Those limits are clearly stated on our website, and while I agree that we need to update this screenshot, finding a single example of an outdated screenshot and saying it warrants jail time is a little excessive. Yes there is a user limit, and we are extremely up front about it. It's even underlined and highlighted on our pricing page, with a tooltip describing exactly what the user limits are.

The Budibase quote you referenced about is being open source is entirely correct. Most of Budibase is GPL v3, but the libraries that run your apps are MPL licensed (even more permissive).
![image](https://github.com/Budibase/budibase/assets/9075550/c3a68a01-e7dd-4bec-848b-c2e189edc28c)

We don't state we are entirely free, and being free is not the same as being open source. We are self-hostable and the majority of Budibase (apart from the paid features) are open source. You're absolutely welcome to fork the code and change it how you see fit, including removing restrictions in the open source code if you're able to do that. The only thing you can't do is access the paid features, which are in a closed source repository and are minified/obfuscated for self-hosted installations.

To address your points directly:
> Requesting the same amount for the self hosted and cloud variant, despite you don't have to pay for others self hosted instances

I've touched on this in previous replies - check my comment about Photoshop for example. We've spent a lot of time and money building the product, and Budibase is providing you with value no matter where it runs. Just because it isn't running on our servers does not mean we deserve no compensation for building and offering this software. Self-hosting is also more commonly a requirement for larger companies with tighter data regulations.

> The misleading advertisement (which is even ILLEGAL in a lot of countries (the one I'm living in as example. You would get fined up to 6 figures here or even jail time for such things))

We will update this screenshot, but as I've already said we do not lie about limits - they are clearly stated on the website which is the actual source of truth for this information rather than a picture buried in the docs.

> This is extremely shady behavior I don't like.

You're entitled to that opinion, but I hope my explanation helps alleviate that somewhat. We absolutely do not attempt to mislead users, and I genuinely don't believe there's any evidence here of shady behaviour. We have a limit on the user count on free plans, and we tell you about it in advance.

> I can't even understand why users (and groups etc) are restricted on self hosted, because you have literally no downsides of it being unlimited.

Because we need to make money as a company somehow, and we choose to do that by encouraging moving to paid plans whenever you have so much success with Budibase that you want to roll it out to more people. It's not about the direct cost to us - It's about striking a balance between offering enough for free to allow users (and indeed companies) to use Budibase successfully, while also ensuring there is eventually some reason to transition to a paid plan. Limits of scale are a very fair way to do this. Groups are again just a convenience feature which provides more benefits at scale - using groups when you might only have 5 users doesn't provide much value, but if you have 500 users then they can save you a lot of time.

**Reply by JCV67** [reply]
Sorry but when I started, and log after that, there where no user limits nor where they announced. You should takes this in consideration. Also, think that MS 365 office suit includs Power Apps and the basic plan price is similar to Budibase's. You should have a different plan for older self hosted users.

**Reply by XeroxDev** [reply]
@aptkingston 
> @XeroxDev you're right that the screenshot you provided is outdated - we'll fix that. But that absolutely is not representative of our "advertising" of plan limits. Those limits are clearly stated on our website, and while I agree that we need to update this screenshot, finding a single example of an outdated screenshot and saying it warrants jail time is a little excessive.

Great! And: I never said due to that screenshot alone it warrants jail time. I said that in my country wrong advertisement may result in fines or jail time. And you clearly advertised it in a lot of places.

Even your Co-Founder "jo_ranamo" went around on reddit and advertised it with "unlimited users, free forever" etc. (But hey. The entire account and all the comments were deleted by the user itself)
![image](https://github.com/Budibase/budibase/assets/10864620/ac214959-6ec9-4903-a6b9-3179d20fb3e2)

> The Budibase quote you referenced about is being open source is entirely correct. Most of Budibase is GPL v3, but the libraries that run your apps are MPL licensed (even more permissive).

That might be correct, but it was advertised as self hosted OSS, free forever. Oh, and the funny part: Setting up the project and using the non-private stuff only results in an unusable instance, because a huge amount of stuff has to be re-implemented.

> We don't state we are entirely free, and being free is not the same as being open source. We are self-hostable and the majority of Budibase (apart from the paid features) are open source. You're absolutely welcome to fork the code and change it how you see fit, including removing restrictions in the open source code if you're able to do that. The only thing you can't do is access the paid features, which are in a closed source repository and are minified/obfuscated for self-hosted installations.

👍 

> I've touched on this in previous replies - check my comment about Photoshop for example. We've spent a lot of time and money building the product, and Budibase is providing you with value no matter where it runs. Just because it isn't running on our servers does not mean we deserve no compensation for building and offering this software. Self-hosting is also more commonly a requirement for larger companies with tighter data regulations.

You can repeat it as often as you like. The PS example is literally the worst and most unrelatable one you could've used. There were so many examples which would've been better and yet you're using a PAID ONLY, CLOSED SOURCE, PREMIUM SERVICE from ADOBE, which you can't even self-host as an example? What?!

> We will update this screenshot, but as I've already said we do not lie about limits - they are clearly stated on the website which is the actual source of truth for this information rather than a picture buried in the docs.

- Great, thanks again! (But you should just recapture the image, instead of just cropping it. The cropped version is kinda unsatisfying :D
- "The actual source of truth" is your website? Got it!
![image](https://github.com/Budibase/budibase/assets/10864620/96d76374-dea4-4ea9-b2e3-1f26e4572dd1)
(Oh. And the "Community Plan" states "FREE FOREVER")
That's just one site of your official page stating it. There're also many others, but this will bloat that post, especially if we would referencing official statements from your Co-Founder and members.
- "picture buried in the docs". Really? This is the first page a user finds if he searches for activating the self hosted version! It's also the ONLY page speaking about that, so it's not buried, it's literally the most important one and IIRC you're even linking it yourself. Also: Burried would imply it beeing multiple pages down and yet it's instant visible and screaming "click me" 
![image](https://github.com/Budibase/budibase/assets/10864620/a38ee865-311e-4924-b12c-ca470a6ab08a)

> You're entitled to that opinion, but I hope my explanation helps alleviate that somewhat. We absolutely do not attempt to mislead users, and I genuinely don't believe there's any evidence here of shady behaviour. We have a limit on the user count on free plans, and we tell you about it in advance.

I believe you that it's not intentional and yet it seems intentional and shady. But thanks for your permission to have other opinions lol. But jokes aside. There would be so many other possibilities to make money and I would even help you find them, because you deserve it, but restricting things which were advertised as "free forever" and making them suddenly non-free. That's why I'm acting like this. You could develop incredible features everyone would love and put them behind a paywall. You could implement QOL features premium only, you could even offer services just for the cloud paid version. Or even make a "pay/donate what you want" license for self hosted. I would be fine with all of it and would even consider giving you money. But you broke the only OSS rule no one should break. Putting something "free forever" behind a paywall.

> Because we need to make money as a company somehow, and we choose to do that by encouraging moving to paid plans whenever you have so much success with Budibase that you want to roll it out to more people. It's not about the direct cost to us - It's about striking a balance between offering enough for free to allow users (and indeed companies) to use Budibase successfully, while also ensuring there is eventually some reason to transition to a paid plan. Limits of scale are a very fair way to do this. Groups are again just a convenience feature which provides more benefits at scale - using groups when you might only have 5 users doesn't provide much value, but if you have 500 users then they can save you a lot of time.

Like mentioned above. I understand, accept and agree that you have to earn money. But also like mentioned above: There would've been other ways to do that.

The groups thing is fine, it was just an example. You can, of course put it behind a paywall. I respect that.

I'm not here to "destroy", insult, other anything else. I just wanted to explain my opinions to you and hope you reconsider putting "free forever" things behind a paywall. (And yes. I would even argue like that if you'd set the limit to 1 million.)

**Reply by aptkingston** [reply]
@XeroxDev the community plan is indeed free forever. That has not changed, and will not change. Imposing a 20 user limit does not change the fact it's free - by definition, charging for it would.

Thanks again for the contribution of the outdated screenshot which has now been rectified. I think this conversation has gone slightly off topic, so I'm going to lock this thread for now as it has served its purpose. We appreciate the feedback, and you're always welcome to create new discussions in future if you have any further questions.

Edit - I think it's also worth mentioning that the page you linked from our website is a blog post from last year and not our up-to-date pricing information, which can always be found on our actual pricing page. But please do let us know if you find inconsistencies in our live content, like the docs image you mentioned previously.
